File: backend/app/schemas/travel.py
from pydantic import BaseModel
from datetime import datetime
from enum import Enum
from typing import Optional

class TravelStatus(str, Enum):
    UPCOMING = "upcoming"
    ONGOING = "ongoing"
    COMPLETED = "completed"

class TravelResponse(BaseModel):
    id: str
    user_id: str
    destination: str
    start_date: datetime
    end_date: datetime
    status: TravelStatus
    image_url: Optional[str] = None
    description: Optional[str] = None
    
    class Config:
        json_schema_extra = {
            "example": {
                "id": "607f1f77bcf86cd799439011",
                "user_id": "507f1f77bcf86cd799439012",
                "destination": "Salvador, Bahia",
                "start_date": "2026-07-15T00:00:00Z",
                "end_date": "2026-07-22T00:00:00Z",
                "status": "upcoming",
                "image_url": "https://...",
                "description": "Viagem de luxo em Salvador"
            }
        }

class TravelListResponse(BaseModel):
    travels: list[TravelResponse]
    count: int

File: frontend_flutter/lib/features/client/domain/entities/travel.dart
import 'package:flutter/foundation.dart';

/// Representa uma viagem do cliente no portal.
/// Mapeia diretamente o schema TravelResponse do backend.
@immutable
class Travel {
  const Travel({
    required this.id,
    required this.userId,
    required this.destination,
    required this.startDate,
    required this.status,
    this.endDate,
    this.imageUrl,
    this.description,
  });

  final String id;
  final String userId;
  final String destination;
  final DateTime startDate;
  final DateTime? endDate;
  final String status;
  final String? imageUrl;
  final String? description;

  bool get isUpcoming => status == 'upcoming';
  bool get isOngoing => status == 'ongoing';
  bool get isCompleted => status == 'completed';

  factory Travel.fromJson(Map<String, dynamic> json) {
    return Travel(
      id: json['id'] as String,
      userId: json['user_id'] as String,
      destination: json['destination'] as String,
      startDate: DateTime.parse(json['start_date'] as String),
      endDate: json['end_date'] != null
          ? DateTime.parse(json['end_date'] as String)
          : null,
      status: json['status'] as String,
      imageUrl: json['image_url'] as String?,
      description: json['description'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'destination': destination,
      'start_date': startDate.toIso8601String(),
      'end_date': endDate?.toIso8601String(),
      'status': status,
      'image_url': imageUrl,
      'description': description,
    };
  }
}
